### DISCUSSION TITLE
Custom logging configuration is skipped

### DISCUSSION THREAD

**Discussion by ivanov-slk** [dagster-io/dagster#4168] [Q&A] [answered]
I am trying to replicate the example given [here](https://docs.dagster.io/concepts/logging/loggers#configuring-the-built-in-loggers). 

However, when I execute the pipeline _without_ the yaml configuration (relying on the defaults), the custom logger function is never used. Which is, if I execute `dagster pipeline execute -f test.py`.

If I use the yaml file and do `dagster pipeline execute -f test.py  -c test.yaml` I see JSON output, as expected. I see correct behavior if I specify the logger configuration via the `run_config = {}` argument in `execute_pipeline`, again as expected.

My expectations were that the custom logger should always be active - it has default values for the necessary configuration (log level and name) and the `ModeDefinition` is always used. 
What am I missing?

I am using dagster 0.11.8 on python 3.6.
My code is as follows - I have only added the necessary imports.
```
from dagster import logger, solid, pipeline, execute_pipeline, Field, ModeDefinition
import logging
import json


@logger(
    {
        "log_level": Field(str, is_required=False, default_value="INFO"),
        "name": Field(str, is_required=False, default_value="dagster"),
    },
    description="A JSON-formatted console logger",
)
def json_console_logger(init_context):
    level = init_context.logger_config["log_level"]
    name = init_context.logger_config["name"]

    klass = logging.getLoggerClass()
    logger_ = klass(name, level=level)

    handler = logging.StreamHandler()

    class JsonFormatter(logging.Formatter):
        def format(self, record):
            return json.dumps(record.__dict__)

    handler.setFormatter(JsonFormatter())
    logger_.addHandler(handler)
    return logger_


@solid
def hello_logs(context):
    context.log.info("Hello, world!")


@pipeline(mode_defs=[ModeDefinition(logger_defs={"my_json_logger": json_console_logger})])
def demo_pipeline():
    hello_logs()
``` 

I have also copied the configuration in the yaml file:
```
loggers:
  my_json_logger:
    config:
      log_level: INFO
```

**Comment by alangenfeld** [comment, marked as answer]
The current set up does require "activating" the logger by at least passing
```
loggers:
  my_json_logger:
```

In the run config. The defaults in the config allow you to skip providing the inner details.

I agree this isn't the most intuitive behavior, and is something we will have in mind as we improve these systems.

**Reply by ivanov-slk** [reply]
Thanks! 

The good thing is that you can do this also by passing `run_config` in `execute_pipeline`, so no need to explicitly use YAML file.
